packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -548,10 +548,14 @@ export default class NewVm extends BaseComponent {
     const defaultNetworkIds = this._getDefaultNetworkIds(template)
     forEach(template.VIFs, vifId => {
       const vif = getObject(storeState, vifId, resourceSet)
-      VIFs.push({
+      const networkObject = {
         network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0],
-      })
+        deviceIndex: vif.device
+      }
+      VIFs.push(networkObject)
     })
+    // sort the array so it corresponds to the order chosen by the user when creating the VM
+    VIFs.sort((a, b) => Number(a.deviceIndex) - Number(b.deviceIndex))
```

Comment:
Keep `device` as `device`:

```suggestion
    VIFs.sort((a, b) => Number(a.device) - Number(b.device))
```